### DISCUSSION TITLE
Welcome to Harbour Discussions!

### DISCUSSION THREAD

**Discussion by vszakats** [harbour/core#220] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by mlacecilia** [comment]
Hi Viktor and all,
I am a Jurassic dBase developer and enthusiastic Harbour user, as well as a very modest contributor.
This community could be a springboard for new language expansions and for setting some standards for its use.
Personally I am very interested in any development aimed at integrating improved multi-platform functions in Harbour, the ability to create web-based applications and, even more stimulating, real development possibilities in an Android or iOS environment.
Although no longer young and retired, I have no intention of abandoning this stimulating language and both the new and the historical friends of the Harbour discussion groups.
PS: many thanks, Viktor, to be back in supporting Harbour!

**Comment by BrynGerard** [comment]
I'm an old time dBaseII and Clipper developer. Very Interested in using Harbour.  The Cross-platform development side of things is a big plus.

**Comment by toninhofwi** [comment]
hi ppl, I'm a 54 years old xBase developer, glad to join in this place and help.

**Comment by lautaromoreira** [comment]
Hello,
I'm a 47 years old Harbour developer, the multiplatform is a great plus of harbour.

**Comment by bedipritpal** [comment]
Hi Viktor

Thanks for setting this chapter for Harbour community. Will greately help to bring all together.

I am a fanatic Harbour user since I joined xHarbour in 2002 than sided with Harbour in 2007 sometime. A 65 year old but turned to software space - Clipper - in 1993 at age 37, a fairly late entrant by any standards. All these years whatever I developed revolves around Harbour, and only Harbour, whatever it be, commercial applications, libraries, etc., on whatever platform, I succeeded. All these years Harbour has remained my passion as well as bread-and-butter.

And this would not been possible without you and Przemek, the two pillers of Harbour, who have immensly rasised the bar with selfless dedication to this project. I pay my respect to both of you specifically, and all those legends who contributed to this project.

Said that, I have a request for both of you - please return to the project and refuel it with your wisdom. It feels orphan without you.

Pritpal Bedi
a student of software analysis & concepts

**Comment by kenny1818** [comment]
Hi Viktor and all,
I'm learning xbase on dBase2 cpm/80,later start developer on clipper s87,xharbour,miniguiExt,harbour,HMG.

**Comment by mbelgrano** [comment]
Hi Viktor
nice to see you here

**Comment by VictorCasajuanaMas** [comment]
Hi Viktor Master!

I'am a 47 years old programmer and coder of dbase and clipper from 1988 aprox. then harbour with fivewin so far.
I've also used other languages like pascal, c, php, javascript
Several years ago I learned other frameworks (Laravel, Angular) and I think that harbour is a great languaje.

**Comment by spiccioli** [comment]
Hi all,

I'm a long time user of harbour on OS/2 and Windows and I've made some 
small contributions to the project in the past.

Regards.

Maurilio.



Il 09/12/2020 12.57, Viktor Szakats ha scritto:

**Comment by MarcioSistema** [comment]
Good afternoon, I’m old too, my first clipper book was summer 87 when I was 13 years old. Today I work with xHarbour and the Visual IDE xHarbour along with MySql in native mode. I have product in linux, windows and all working on cloud computer, unfortunately to meet customer with mobile work today with Windev.

**Comment by diegopego** [comment]
Hi Viktor! Welcome back! We sure missed you a lot!
I am Diego. I work with Harbour since 2010.
I do the heavy lifting using hb 3.0 and DBF (lots of legacy code without tests.) All my interface is Text based UI via SSH.
I am still learning and I'm eager to contribute to the community.
Currently, I'm creating a pure and simple harbour xUnit test framework based inspired by CppUTest and Unity. All of its tests are written using the library itself.
Like the great Pritpal said, this Harbour has been my bread-and-butter, and passion since then. It has almost all the features desired in a language. Most, if not all of the "modern" languages should be learning from it.

**Comment by microvolution** [comment]
Olá Viktor! tudo bem?
o que significa ou como usar no celular/android o harbour? já que você coloca no hbmk2 as seguintes observações:
****
A plataforma de destino {unix} é compatível com * nix (bsd, hpux, sunos, beos, qnx, android, vxworks, symbian, linux, darwin, cygwin, minix, aix)
****
android - Android
****
android
gcc - GNU C x86
gccarm - GNU C ARM
****
 x | win      | android/gcc       | x86
 x | win      | android/gccarm    | arm
****
 x | linux    | android/gcc       | x86
 x | linux    | android/gccarm    | arm
****
 x | darwin   | android/gcc       | x86
 x | darwin   | android/gccarm    | arm
****
****

Ou seja, na sua página inicial https://github.com/harbour/core, aparecem 9 vezes a palavra android.

e no help do HBMK2 -HELP, também oferece o seguinte, que inclusive vem assinado por você e fala sobre android (gcc arm):
Valores suportados para <compiler> conforme a <platform> disponível:
*********
  linux    gcc, clang, icc, watcom, sunpro, open64
  darwin   gcc, clang, icc
  win      mingw, msvc, clang, bcc, bcc64, watcom, icc, pocc, xcc, mingw64,
           msvc64, msvcia64, iccia64, pocc64
  wce      mingwarm, mingw, msvcarm, poccarm
  os2      gcc, gccomf, watcom
  dos      djgpp, watcom
  bsd      gcc, clang, pcc
  hpux     gcc
  beos     gcc
  qnx      gcc
  android  gcc, gccarm
  vxworks  gcc, diab
  symbian  gcc
  cygwin   gcc
  minix    clang, gcc
  aix      gcc
  sunos    gcc, sunpro

Autor:

  Viktor Szakáts (vszakats.net/harbour)
*********

ENFIM, COMO USAR O HARBOUR DENTRO DE UM CELULAR sem precisar de JAVA, JAVASCRIPT, HTML, CSS, ou outros softwares intermediários?

**Comment by CCH4GITHUB** [comment]
Hi Viktor

Nice to see you here :-) I am Chee from Penang, Malaysia and use Harbour for my 32/64 bits report writing and processing engines behind my Delphi Windows apps.
